plugins/animate/squeezimize.hpp

```diff
@@ -0,0 +1,388 @@
+                    self->minimize_target.width),
+                    (self->minimize_target.x + self->minimize_target.width) - src_box.x),
+                    (src_box.x + src_box.width) - self->minimize_target.x);
+            self->animation_geometry.height =
+                std::max(std::max(std::max(src_box.height,
+                    self->minimize_target.height),
+                    (self->minimize_target.y + self->minimize_target.height) - src_box.y),
+                    (src_box.y + src_box.height) - self->minimize_target.y);
+
+            const float vertex_data_pos[] = {
+                1.0f * self->animation_geometry.x,
+                1.0f * self->animation_geometry.y + self->animation_geometry.height,
+                1.0f * self->animation_geometry.x + self->animation_geometry.width,
+                1.0f * self->animation_geometry.y + self->animation_geometry.height,
+                1.0f * self->animation_geometry.x + self->animation_geometry.width,
+                1.0f * self->animation_geometry.y,
+                1.0f * self->animation_geometry.x, 1.0f * self->animation_geometry.y,
+            };
+
+            const glm::vec4 src_box_pos{
+                float(src_box.x - self->animation_geometry.x) / self->animation_geometry.width,
+                float(src_box.y - self->animation_geometry.y) / self->animation_geometry.height,
+                float((src_box.x - self->animation_geometry.x) + src_box.width) /
+                self->animation_geometry.width,
+                float((src_box.y - self->animation_geometry.y) + src_box.height) /
+                self->animation_geometry.height
+            };
+
+            const glm::vec4 target_box_pos{
+                float(self->minimize_target.x - self->animation_geometry.x) / self->animation_geometry.width,
+                float(self->minimize_target.y - self->animation_geometry.y) / self->animation_geometry.height,
+                float((self->minimize_target.x - self->animation_geometry.x) + self->minimize_target.width) /
+                self->animation_geometry.width,
+                float((self->minimize_target.y - self->animation_geometry.y) + self->minimize_target.height) /
+                self->animation_geometry.height
+            };
+
+            OpenGL::render_begin(target);
+            self->program.use(wf::TEXTURE_TYPE_RGBA);
+            self->program.uniformMatrix4f("matrix", target.get_orthographic_projection());
+            self->program.attrib_pointer("position", 2, 0, vertex_data_pos);
+            self->program.attrib_pointer("uv_in", 2, 0, vertex_data_uv);
+            self->program.uniform1i("upward", upward);
+            self->program.uniform1f("progress", progress);
+            self->program.uniform4f("src_box", src_box_pos);
+            self->program.uniform4f("target_box", target_box_pos);
+            self->program.set_active_texture(src_tex);
+            GL_CALL(glDrawArrays(GL_TRIANGLE_FAN, 0, 4));
```

Comment:
Seems like this is lacking scissoring to the damaged region. It probably is not noticeable when the animated view is on top, but it may be causing issues if for example the animated view ends up behind another opaque view which is blocking part of it.

Reply:
Simplest way to do it would be just iterating over the damage rects and calling glDrawArrays multiple times (core does this).

Reply:
Yes, sounds like it should work, so long as the pre_hook is in place to damage the full region.